### PR TITLE
SDK: Configurator

### PR DESCRIPTION
**Pull request by acaldas**
Related to: https://github.com/ripe-tech/ripe-core/issues/2228

### CODE REVIEW

File: src/js/visual/config.part.js [hunk 1]
```diff
@@ -6,17 +6,550 @@ ripe.Config = function(owner, element, options) {
...
+    back.height = size;
+    back.style.marginLeft = "-" + String(size) + "px";
+    mask.width = size;
+    mask.height = size;
+    this.element.setAttribute("data-current-size", size);
+    this.update();
+};
+
+ripe.Config.prototype.update = function(state, options) {
+    if (this.ready == false) {
```
Comment (hugo-gomes) [inline src/js/visual/config.part.js]: use === instead

File: src/python/ripe_demo/static/js/main.js [hunk 2]
```diff
@@ -55,6 +55,10 @@ window.onload = function() {
         frame: "top"
     });
 
+    setTimeout(function() {image.setFrame("9")});
```
Comment (hugo-gomes) [inline src/python/ripe_demo/static/js/main.js]: missing semi colon

File: src/python/ripe_demo/templates/partials/layout.demo.html.tpl [hunk 3]
```diff
@@ -18,10 +18,10 @@
         <div id="canvas" class="canvas" data-url="{{ url|default('', True) }}"
              data-brand="{{ brand|default('', True) }}" data-model="{{ model|default('', True) }}"
              data-variant="{{ variant|default('', True) }}" data-country="{{ country|default('', True) }}"
-             data-currency="{{ currency|default('', True) }}">
-            <img id="frame-0" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" />
-            <img id="frame-6" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" />
-            <img id="frame-top" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" />
+             data-currency="{{ currency|default('', True) }}" data-size="620" data-max_size="620" data-sensitivity="40" data-position="0" data-view="side">
         </div>
+        <img id="frame-0" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" />
```
Comment (joamag) [inline src/python/ripe_demo/templates/partials/layout.demo.html.tpl]: why this change ?

File: src/js/visual/visual.part.js [hunk 4]
```diff
@@ -1,12 +1,51 @@
...
     ripe.Observable.call(this);
+    ripe.Interactable.call(this, owner, options);
 
     this.element = element;
     ripe.Visual.prototype.init.call(this);
 };
 
-ripe.Visual.prototype = Object.create(ripe.Interactable.prototype);
 ripe.Visual.prototype = Object.create(ripe.Observable.prototype);
+ripe._assign(ripe.Visual.prototype, ripe.Interactable.prototype);
```
Comment (joamag) [inline src/js/visual/visual.part.js]: maybe we should promote assign to assign with no _

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+    "env": {
+        "browser": true
+    },
+    "extends": "eslint:recommended",
+    "rules": {
+        "indent": [
```
Comment (joamag) [inline .eslintrc.json]: "indent": ["warn", 4],
        "quotes": ["error", "double"],
        "semi": ["error", "always"],
        "space-before-function-paren": ["error", "never"],
        "mocha/no-exclusive-tests": "error"


add this extra rules

File: src/js/base/api.js [hunk 6]
```diff
@@ -1,40 +1,58 @@
...
+        frames[face] = 1;
+    };
+
+    var sideFrames = this.config["frames"];
+    frames["side"] = sideFrames;
+    this.frames = frames;
+    callback && callback(frames);
+};
+
+ripe.Ripe.prototype._requestUrl = function(url, callback) {
```
Comment (joamag) [inline src/js/base/api.js]: change the name of the function to _requestURL

File: src/css/ripe.css [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+body .content .configurator .mask {
+    display: none;
+}
+
+body .content .configurator .masks {
+    display: none;
+}
+
+body .content .configurator.fullscreen {
```
Comment (joamag) [inline src/css/ripe.css]: don't use body content just go with configurator.fullscreen

File: src/js/visual/config.part.js [hunk 8]
```diff
@@ -1,22 +1,601 @@
...
 ripe.Config.prototype = Object.create(ripe.Visual.prototype);
 
 ripe.Config.prototype.init = function() {
-    this.owner.addSelectedPartCallback(function(part) {
+    this.size = this.element.dataset.size || this.options.size || 1000;
+    this.maxSize = this.element.dataset.max_size || this.options.maxSize || 1000;
+    this.sensitivity = this.element.dataset.sensitivity || this.options.sensitivity || 40;
+    this.duration = this.options.duration || 0;
+
+    this.owner.bind("selected_part", function(part) {
```
Comment (joamag) [inline src/js/visual/config.part.js]: move this to the last part of the function

File: src/js/visual/config.part.js [hunk 9]
```diff
@@ -1,22 +1,601 @@
...
+ripe.Config.prototype.enterFullscreen = function(options) {
+    if (this.element === undefined) {
+        return;
+    }
+    this.element.classList.add("fullscreen");
+    var maxSize = options.maxSize || this.element.dataset.max_size || this.maxSize;
+    this.resize(maxSize);
+};
+
+ripe.Config.prototype.exitFullscreen = function(options) {
```
Comment (joamag) [inline src/js/visual/config.part.js]: leaveFullscreen

File: src/js/visual/config.part.js [hunk 10]
```diff
@@ -1,22 +1,601 @@
...
+    var elementWidth = this.element.clientWidth;
+    var elementHeight = this.element.clientHeight || child.clientHeight;
+    var percentX = deltaX / elementWidth;
+    var percentY = deltaY / elementHeight;
+    this.element.dataset.percent = percentX;
+    var sensitivity = this.element.dataset.sensitivity || this.sensitivity;
+
+    // if the movement was big enough then
+    // adds the move class to the element
+    Math.abs(percentX) > 0.02 && this.element.classList.add("move");
```
Comment (joamag) [inline src/js/visual/config.part.js]: the 0.02 paramter must be a paramter

File: src/js/visual/config.part.js [hunk 11]
```diff
@@ -1,22 +1,601 @@
...
+    Math.abs(percentY) > 0.02 && this.element.classList.add("move");
+
+    // if the drag was vertical then alters the
+    // view if it is supported by the product
+    var view = this.element.dataset.view;
+    var nextView = view;
+    if (sensitivity * percentY > 15) {
```
Comment (joamag) [inline src/js/visual/config.part.js]: the 15 must be a parameter (threshold)
